cmd/services.rb

```diff
@@ -40,11 +40,19 @@ def services_args
   def services
     args = services_args.parse
 
-    raise UsageError, "`brew services` is supported only on macOS!" unless OS.mac?
-
     # Keep this after the .parse to keep --help fast.
     require_relative "../lib/services_cli"
 
+    if !ServicesCli.launchctl? && !ServicesCli.systemctl?
+      raise UsageError,
+            "`brew services` is supported only on macOS or SystemD!"
+    end
+
+    if ServicesCli.systemctl?
+      uid = Utils.safe_popen_read("id", "--user").chomp
+      ENV["DBUS_SESSION_BUS_ADDRESS"] = "unix:path=/run/user/#{uid}/bus"
```

Comment:
Could you explain what this does?

Reply:
Systemd needs this variable and brew filters it. So I set it up again. This might not work in all cases but I don't know a better way.

Reply:
So apparently that doesn't work in the container. I'm going to need a new idea here. Can I somehow unfilter variables for specific brew commands?

Reply:
Yeh, this is possible but at what point does it need to be set?

Reply:
It needs to be available for any interaction with SystemD

Reply:
This should be doing it then. Are you sure it's being set correctly and not e.g. missing other variables?